src/trainer_see.c

```diff
@@ -474,7 +474,7 @@ static u8 CheckTrainer(u8 objectEventId)
         if (GetTrainerFlagFromScriptPointer(trainerBattlePtr))
         {
             //If there is a rematch, we want to trigger the approach sequence
-            if (GetRematchFromScriptPointer(trainerBattlePtr))
+            if (GetRematchFromScriptPointer(trainerBattlePtr) && I_VS_SEEKER_CHARGING)
```

Comment:
```suggestion
            if (I_VS_SEEKER_CHARGING && GetRematchFromScriptPointer(trainerBattlePtr))
```
Configs always go first so the compiler can optimise them out